### ISSUE TITLE
/droprate herbert(pet) giving wrong .txt file

### ISSUE THREAD

**Issue by TURB042O** [oldschoolgg/oldschoolbot#5265]
# title says all
- here are images 

![SmartSelect_20230815_114259_Discord.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/382d3f2a-d5b7-4bbe-bb8f-9647f282f515)

![SmartSelect_20230815_114310_Discord.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/bb1ee7dd-43f3-4bad-947e-229634dbea87)

![I am sharing 'droprates' with you_230815_114751_1.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/8e2200d6-5922-4ab8-b160-f60effa9b464)

![I am sharing 'droprates' with you_230815_114751_2.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/1ac46e8f-fc56-40d6-b214-341072672b2a)

![I am sharing 'droprates' with you_230815_114751_3.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/4fd6bd49-5dd7-41c3-ac28-020a942e3991)

![I am sharing 'droprates' with you_230815_114751_4.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/aa9a8ee4-10dd-458c-8cf0-813118e88384)

![I am sharing 'droprates' with you_230815_114751_5.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/48ac457b-a4eb-4757-aabd-2956a42b952d)

### this is how it should look
![SmartSelect_20230815_115127_Discord.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/951546b8-c8ab-4934-a088-b6548f2ec564)

**Comment by LuxrayElite** [comment]
TLDR: Working as intended

So looking at the code this is due to having a mechanic related to the level of the potion as well as rolling per minute therefore each potion has a unique drop rate.

The text file is due to it exceeding discord character limit for a regular message (2500 or 5000)

The same is true of Baba yaga house pet

Source: 
https://github.com/oldschoolgg/oldschoolbot/blob/b2a273f785c57bcb21ad01b0d99931d1da0e2767/src/mahoji/commands/droprates.ts#L46-L61

Function:
https://github.com/oldschoolgg/oldschoolbot/blob/b2a273f785c57bcb21ad01b0d99931d1da0e2767/src/lib/constants.ts#L791-L797C2

**Comment by TURB042O** [comment]
> TLDR: Working as intended
> 
> So looking at the code this is due to having a mechanic related to the level of the potion as well as rolling per minute therefore each potion has a unique drop rate.
> 
> The text file is due to it exceeding discord character limit for a regular message (2500 or 5000)
> 
> The same is true of Baba yaga house pet
> 
> Source: 
> https://github.com/oldschoolgg/oldschoolbot/blob/b2a273f785c57bcb21ad01b0d99931d1da0e2767/src/mahoji/commands/droprates.ts#L46-L61
> 
> Function:
> https://github.com/oldschoolgg/oldschoolbot/blob/b2a273f785c57bcb21ad01b0d99931d1da0e2767/src/lib/constants.ts#L791-L797C2
> 
> 

well that now makes more sense. 

now that is explained, ![SmartSelect_20230815_114259_Discord.jpg](https://github.com/oldschoolgg/oldschoolbot/assets/133211494/d828d5bd-702e-4797-99e6-60fb0db3ddcf)

maybe this would be more beneficial?  and cause less confusion. just a thought

**Comment by LuxrayElite** [comment]
All file's outputted have the generic name due to being generated by a function related to the command that calls it. EG bank.txt can be any combination of additional filters/searches/sorts using ``/bank format:text_Full``

Generally text files only seem to be an issue on mobile due to having to download them to see any contents.

It would be nice but then it adds a lot of conditional statements to code each text file name🤷
